### PR TITLE
[10.0][IMP] account_financial_report_qweb: Improve performances

### PR DESCRIPTION
**Pull request by rousseldenis**
We rely on full reconciliation and remove the HAVING clause
which could be bulky as operating after the SELECT.

### CODE REVIEW

**Comment by rousseldenis** [comment]
WIP

**Comment by rousseldenis** [comment]
@sbidoul @HviorForgeFlow @JordiBForgeFlow @jcoux 

We experienced problems with aged balance especially with great number of unreconciled items.

File: account_financial_report_qweb/report/open_items.py [hunk 1]
```diff
@@ -428,27 +425,20 @@ def _get_line_sub_query_move_lines(self,
...
         """
         sub_query += """
+            LEFT JOIN account_full_reconcile afr ON afr.id = ml.full_reconcile_id
             WHERE
                 ra.report_id = %s
+            AND ml.full_reconcile_id IS NULL OR afr.create_date >= %s
             GROUP BY
                 ml.id,
                 ml.balance,
                 ml.amount_currency
```
Comment (sbidoul) [inline account_financial_report_qweb/report/open_items.py:441]: IMO, this GROUP BY can be removed. GROUP BY id does not make much sense to me, and the id should be unique across the two parts of the union.

File: account_financial_report_qweb/report/open_items.py [hunk 2]
```diff
@@ -617,17 +607,19 @@ def _inject_line_values(self, only_empty_partner_line=False):
...
-             self.date_at,
+             full_reconcile_date,
              self.date_at,
              self.id,
+             full_reconcile_date,
              self.env.uid,
              self.id,
-             self.date_at,)
+             self.date_at,
+             )
```
Comment (sbidoul) [inline account_financial_report_qweb/report/open_items.py:623]: Just a side note (no need to change): in such  big queries, using `%(name)s` and passing a dict to `execute` is much easier to comprehend and maintain.

File: account_financial_report_qweb/report/open_items.py (hunk 1 above)
Comment (sbidoul) [reply account_financial_report_qweb/report/open_items.py:441]: Hm, no I retract this. There are these LEFT JOIN that repeat the id.

File: account_financial_report_qweb/report/open_items.py [hunk 3]
```diff
@@ -428,27 +425,20 @@ def _get_line_sub_query_move_lines(self,
...
             LEFT JOIN
                 account_move_line ml_past
                     ON ml.balance > 0 AND pr.credit_move_id = ml_past.id
                     AND ml_past.date <= %s
         """
         sub_query += """
+            LEFT JOIN account_full_reconcile afr ON afr.id = ml.full_reconcile_id
             WHERE
                 ra.report_id = %s
+            AND ml.full_reconcile_id IS NULL OR afr.create_date >= %s
```
Comment (sbidoul) [inline account_financial_report_qweb/report/open_items.py:437]: nit: indent this 4 chars

**Review by sbidoul — APPROVED** [review]
Looking for ml that are either not fully reconciled, or have been fully reconciled after the pivot date is apparently faster way to reduce the search space. The amounts will be the same as these still consider the move lines before or on the pivot date.

**Review by HviorForgeFlow — APPROVED** [review]
Performance is quite better! Thanks

**Review by AaronHForgeFlow — APPROVED** [review]
:+1: Much faster in actual database and same result!

File: account_financial_report_qweb/report/open_items.py (hunk 3 above)
Comment (AaronHForgeFlow) [reply account_financial_report_qweb/report/open_items.py:437]: I think this is the only thing pending

Comment (sbidoul) [reply account_financial_report_qweb/report/open_items.py:437]: It's a nit, non blocking for me :)

**Pull request merged**